client/alloc_endpoint.go

```diff
@@ -343,3 +357,21 @@ func (s *execStream) Recv() (*drivers.ExecTaskStreamingRequestMsg, error) {
 	err := s.decoder.Decode(&req)
 	return &req, err
 }
+
+func validateActionExists(actionName string, taskName string, alloc *nstructs.Allocation) (*nstructs.Action, error) {
+	t := alloc.LookupTask(taskName)
+
+	var action *nstructs.Action
+	for _, act := range t.Actions {
+		if act.Name == actionName {
+			action = act
+			break
+		}
+	}
+
+	if action == nil {
+		return nil, fmt.Errorf("action %s not found", action)
+	}
+
+	return action, nil
+}
```

Comment:
Typically we return early when possible from a loop like this. That lets you reduce it to something like:

```suggestion
	for _, action := range t.Actions {
		if action.Name == actionName {
			return action, nil
		}
	}

	return nil, fmt.Errorf("action %s not found", actionName)
}
```

(Plus fixed a bug in the error return value, which would have always been `"action <nil> not found"`)